engine/src/main/scala/cromwell/engine/workflow/lifecycle/materialization/MaterializeWorkflowDescriptorActor.scala

```diff
@@ -306,7 +306,16 @@ class MaterializeWorkflowDescriptorActor(serviceRegistryActor: ActorRef,
       validatedNamespace <- buildValidatedNamespace(factory, sourceAndResolvers._1, sourceAndResolvers._2)
       _ = pushNamespaceMetadata(validatedNamespace)
       ewd <- fromEither[IO](buildWorkflowDescriptor(id, validatedNamespace, workflowOptions, labels, conf, pathBuilders, outputRuntimeExtractor).toEither)
-    } yield ewd
+    } yield {
+
+      // Debugging #4117 - make sure we are creating an importable directory out of the zip (AEN 2018-11-14)
+      zippedImportResolver match {
```

Comment:
I would move this up right below `zippedImportResolver <- zippedResolverCheck`.

Otherwise if something subsequently fails building `ewd` this `yield` block won't run and this log will be lost. And I believe that that's precisely the case when we want to get this logging?

Reply:
That is... an excellent point